### PR TITLE
HPCC-31972 Update Docs for Cost Tracking improvements

### PR DESCRIPTION
**Pull request by g-pan**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Successful Unit Tests : https://github.com/g-pan/github-action-dev-build/actions/runs/10474453301

### CODE REVIEW

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 1]
```diff
@@ -988,11 +1007,17 @@ thor: []
...
+<itemizedlist>
+<listitem>File Access Cost
+</listitem>     
+<listitem>File Cost at Rest
+</listitem>
+</itemizedlist>
         <para>File access costs are the costs of reading and writing to the
         files. Many storage planes do have a separate charge for data
-        operations. The cost of reading and writing to the file will be
+        operations. The cost of reading and writing to the file are
```
Comment (JamesDeFabia) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:1020]: SV agreement. Either "costs are included" or "costs is included"

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 2]
```diff
@@ -865,14 +865,17 @@ thor: []
     using cloud native HPCC Systems platform instances in a cloud setting
     where some planning and configuration can help reduce expenses.</para>
 
-    <para>Two new columns have been added to the workunits page in ECL Watch.
-    The columns may be sorted by either cost column, just like the other
-    columns in ECL Watch, by clicking at the top of the column.</para>
-
+    <para>New columns have been added to the workunits and the logical files pages in ECL Watch.
+    These columns may be sorted by any cost columns, just like the other
```
Comment (JamesDeFabia) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:869]: Use "can" instead of "may"

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
comments inline

**Comment by g-pan** [comment]
@shamser please verify the information going into the doc is currently accurate.

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 3]
```diff
@@ -865,14 +865,17 @@ thor: []
...
-    The columns may be sorted by either cost column, just like the other
-    columns in ECL Watch, by clicking at the top of the column.</para>
-
+    <para>New columns have been added to the workunits and the logical files pages in ECL Watch.
+    These columns can be sorted by any cost columns, just like the other
+    columns in ECL Watch, by clicking at the top of the column.
+    In addition, file operation costs executed by workunits are provided in the workunit’s metrics. </para>
+<para> 
+As the cost tracking matures, the cost calculations also improve significantly providing more accurate cost tracking.  
+For example, data that was accessed from page cache doesn’t incur any file access cost.  HPCC Systems cost tracking now detects data that has been returned from the page cache and adjust cost calculations appropriately resulting in more accurate cost calculations.  </para>
```
Comment (JamesDeFabia) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:874]: s/b "adjusts" for SV agreement

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
one minor comment inline

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 4]
```diff
@@ -882,18 +885,26 @@ thor: []
...
+         <listitem>
+            <para>Compile Cost</para>
+          </listitem>
 
           <listitem>
             <para>File Access Cost</para>
           </listitem>
+          
+          <listitem>
+            <para>File Cost at Rest</para>
```
Comment (shamser) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:897]: Storage Cost is listed as a separate item in the list of costs above.  Storage cost isn't a separate cost item. Storage Cost = File Access Cost + File Cost At Rest.  Also, the Execution cost is not listed.  The Types of costs tracked are:
- Execute Cost
- Compile Cost
- File Access Cost
- File Cost At Rest

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 5]
```diff
@@ -988,31 +1007,40 @@ thor: []
       </sect3>
 
       <sect3 id="FileAccessCosts">
-        <title>File Access Cost</title>
+        <title>File Access Costs</title>
+<para> For logical files, the costs calculated are based on two types of file access.  </para>
```
Comment (shamser) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:1011]: It should probably be "...there are two types of file costs:".  There is just one type of file access cost (File Access Cost) but there are 2 storage related costs (File Access Cost and File Cost at Rest)

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 6]
```diff
@@ -988,31 +1007,40 @@ thor: []
       </sect3>
 
       <sect3 id="FileAccessCosts">
-        <title>File Access Cost</title>
+        <title>File Access Costs</title>
```
Comment (shamser) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:1010]: The title should be something like "File Cost".  File Access Costs doesn't include Storage At Rest cost.

Also, the paragraph on line 1003-1007, is no longer correct. The two types of storage costs has been separated into 2 different fields and can now be viewed as separate fields.

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 7]
```diff
@@ -988,31 +1007,40 @@ thor: []
...
         <para>File access costs are the costs of reading and writing to the
         files. Many storage planes do have a separate charge for data
-        operations. The cost of reading and writing to the file will be
+        operations. The cost of reading and writing to the file is
         included in the file access cost value. Any other cost associated with
         file operations (such as delete or copy) will not be tracked or
         included as part of file access cost at this time.</para>
-
+        
         <para>The file access cost displays as part of the cost field on the
```
Comment (shamser) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:1027]: There are now two separate fields for the file costs now.  The file access costs isn't displayed _as part of_ the cost field.  File access cost is shown as a field. The At rest is shown as separate field.

File: docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml [hunk 8]
```diff
@@ -988,31 +1007,40 @@ thor: []
...
 
-        <para>The new cost field is shown in the Logical File summary page. It
-        is the combined cost of storing and accessing the data.</para>
+      <sect4 id="FileCostAtRest"> <title>File Cost at Rest</title>
 
-        <para>The cost information is currently only generated for Thor and
-        hThor jobs.</para>
-      </sect3>
+        <para>The File Cost at Rest field is shown in the Logical File summary page. It
+        is the cost of storing the file without accessing the data. Only the storage costs associated with housing the file in the cloud. This value has been added to better differentiate between file storage costs. </para>
```
Comment (shamser) [inline docs/EN_US/ContainerizedHPCC/ContainerizedMods/CustomConfig.xml:1041]: "File Cost at Rest field" is the file storage costs.  I think the last line in the previous paragraph should be removed.  File Costs At Rest and File Access cost are both shown in the logical file's summary page.

**Review by shamser — CHANGES_REQUESTED** [review]

**Comment by g-pan** [comment]
provided PDF file for review offline.

**Review by shamser — APPROVED** [review]
Looks good.

**Review by JamesDeFabia — APPROVED** [review]
Good from my POV

**Comment by ghalliday** [comment]
@g-pan please squash

**Review by shamser — APPROVED** [review]

**Pull request merged**